packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -581,11 +609,31 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
                 // (i.e. it has a base mapping) - then we go ahead and snapshot
                 return isAttached;
             }).map(async ([key, value]) => {
-                const channelSummary = await value.summarize(fullTree, trackState);
-                builder.addWithStats(key, channelSummary);
+                const contextSummary = await value.summarize(fullTree, trackState);
+                builder.addWithStats(key, contextSummary);
+
+                // back-compat 0.30 - Older versions will not return references. Set it to empty array.
+                if (contextSummary.references === undefined) {
+                    contextSummary.references = [];
+                }
+                // Add the context's Fluid object references to the list of this channel's references.
+                references.push(...contextSummary.references);
             }));
 
-        return builder.getSummaryTree();
+        // Normalize all the context's path of Fluid object references and prefix them with this channel's id.
```

Comment:
Some comment here would be useful to describe higher-level intention, as it took me a while to get it.
The intention is that we create (for now) circular dependencies between data store and channels, such that they are referenced or collected all at once. In future, we may change that to allow individual DDS (or even maybe sub-DDS) collection.

Reply:
Actually, that's not what this is doing. This is just adding its "id" as a prefix to the path of child's node.
For example, let's say this data store's id is "default" and its child is a DDS "foo".
The nodes returned by "foo" will look something like this -
{ path: "foo", routes: ["/dataStore2"] }
This line will update the path by adding its id as prefix, so this becomes:
{ path: "default/foo", routes: ["/dataStore2"] }

Basically, each layer adds its id as prefix to the path of its children nodes before returning. This will make sure that that "path" we get at the root is the full path.

Reply:
The "addRouteToReferences" call below adds a back reference to itself for all the children so that if a child is referenced, so is this parent.

Reply:
Updated the code and added comments to make this clearer.